src/http.rs

```diff
@@ -26,16 +28,51 @@ pub fn get_client() -> Client<Connector, hyper::Body> {
   Client::builder().build(c)
 }
 
+struct FetchedBodyFuture {
+  body: Concat2<hyper::Body>,
+  status: hyper::StatusCode,
+}
+
+struct FetchedBody {
+  body: hyper::Chunk,
+  status: hyper::StatusCode,
+}
+
+impl Future for FetchedBodyFuture {
+  type Item = FetchedBody;
+  type Error = hyper::Error;
+  fn poll(&mut self) -> Result<Async<FetchedBody>, hyper::Error> {
+    match self.body.poll()? {
+      Async::Ready(body) => Ok(Async::Ready(FetchedBody {
+        body,
+        status: self.status.clone(),
+      })),
+      Async::NotReady => Ok(Async::NotReady),
+    }
+  }
+}
```

Comment:
This seems not so much related to the current problem - although something like this may be necessary at some point when we address fetch body streaming. I'd rather you not add this here. Either go back to using `block_on` twice or find some other way of doing this. Have you tried using [`map()`](https://docs.rs/futures/0.2.1/futures/future/trait.FutureExt.html#method.map) ?